pkg/generator/ingress_translator.go

```diff
@@ -174,6 +177,8 @@ func (translator *IngressTranslator) translateIngress(ctx context.Context, ingre
 			if len(wrs) != 0 {
 				routes = append(routes, envoy.NewRoute(
 					pathName, matchHeadersFromHTTPPath(httpPath), path, wrs, 0, httpPath.AppendHeaders, httpPath.RewriteHost))
+				tlsRoutes = append(tlsRoutes, envoy.NewRoute(
```

Comment:
Do we really want to always do this, even if the Ingress has no TLS config?

Reply:
We don't need it. Updated the PR to create tlsRoutes only when ingress has TLS config.
There are some extra `if` conditions so I always created tlsRoutes in the first PR.